packages/core/src/use-query.ts

```diff
@@ -47,10 +47,14 @@ export function useQuery<T extends QueryFn, O extends Options<T>>(
   }
 
   if (options?.paginated) {
+    // TODO: this could be an issue, unless we assume the `paginated` option will always be constant when it's called
```

Comment:
I refrained from touching this. I can disable this rule, but IMO this is rule we should have to avoid issues later. I think the best solution would be to have `useQuery` and `usePaginatedQuery` as they are different enough ways to query data

Reply:
This will likely not change but it certainly can and should be addressed as it is a bug. Hmm to refactor this to be hooks safe without changing API we could run both and switch out a noop as queryFunction an also set `manual` to `true` on the one we don't want.

Reply:
We can make breaking changes if needed, but I really want the unified API if at all possible.

We can ignore this on this PR but open an issue to discuss a real fix?

Reply:
unless we can guarantee that `paginated` is a constant (maybe we expect the query to have a `paginated` static field rather than a prop to the hook?) react will break if the flag ever changes. When the hooks called during render changes between renders React throws an error which will cause everything within the nearest error boundary to unmount from the page

Reply:
Easy approach would be to not support this changing — throw an error if it does.

We can also fork the react-query code if needed

Reply:
https://github.com/blitz-js/blitz/issues/334